vm/x86/x86defs/src/vmx.rs

```diff
@@ -13,8 +13,8 @@ use zerocopy::IntoBytes;
 use zerocopy::KnownLayout;
 
 open_enum! {
-    /// VMX exit reason
-    pub enum VmxExit: u32 {
```

Comment:
How did we get away with treating this as a full u32 before? Are the various extra bits never set in practice? Do we need to handle them?

Reply:
looking at the tdx module code, they only really send us this bit for vmx enter failure: https://github.com/intel/tdx-module/blob/291112dc72907e9d22d662b29b48d5ba2eca5bdd/src/td_dispatcher/tdx_td_l2_dispatcher.c#L65

which is why we got away with it before. The other bits I'm not actually sure about...

Reply:
filed #1096 to followup